### ISSUE TITLE
Cannot use config.util.setModuleDefaults with dot notation

### ISSUE THREAD

**Issue by bertho-zero** [lorenwest/node-config#497]
### I'm submitting a feature request

As `config.get` support dot notation `config.util.setModuleDefaults` should support it too.

### What is the current behavior?
In my submodule I have:
```js
const moduleConfig = {
  // ...
};

function init(c, moduleName) {
  config.util.extendDeep(moduleConfig, c);

  if (typeof moduleName === 'string') {
    config.util.setModuleDefaults( moduleName, moduleConfig );
  }
}
``` 
And in my main module:
```js
users.init( config.services.users, 'services.users' );
```
Result
```
services: {
  users: {
    paginate: [Object],
    knex: [Object],
    mysql: [Object],
    schemas: [Object],
    files: [Object]
  }
},
'services.users': {
  paginate: [Object],
  knex: [Object],
  mysql: [Object],
  schemas: [Object],
  files: [Object],
  interfaces: [Array]
},
```

### What is the expected behavior?
```
services: {
  users: {
    paginate: [Object],
    knex: [Object],
    mysql: [Object],
    schemas: [Object],
    files: [Object],
    interfaces: [Array]
  }
},
```

### Please tell us about your environment:

 -  node-config version: 2.0.1
 -  node-version: 8.11.3

**Comment by bertho-zero** [comment]
It's done with #498

**Comment by bertho-zero** [comment]
@lorenwest Can you look at this ?

**Comment by lorenwest** [comment]
This looks good. Thanks for the improvement.

**Comment by lorenwest** [comment]
Published today in 2.0.2
